jetty-maven/jetty-maven-core/src/main/java/org/eclipse/jetty/maven/MavenWebAppContext.java

```diff
@@ -0,0 +1,236 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.maven;
+
+import java.io.File;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.jetty.util.resource.CombinedResource;
+
+/**
+ * MavenWebAppContext
+ *
+ * Extends the WebAppContext to specialize for the maven environment. We pass in
```

Comment:
the copy is done automatically so it's not the problem.
Reducing the surface (e.g access to all fields/methods of `WebAppContext`) a  is a bit on purpose (`by design`).
The idea is to have now only access to a limited set of fields because between ee9, ee10 and future eeX those fields may change and we want to give access to only important fields (and this give us more freedom to change `WebAppContext` without breaking users).
Bear in mind this Maven plugin is used for development only and if users have complicated setup they will need to put it in a `jetty-web.xml` file for production so they can use and validate the file with `jetty-maven-plugin` rather than duplicate the content in both `pom.xml` and `jetty-web.xml`. (and that's what we recommend in some javadoc)

Reply:
It's a design choice certainly, but that hasn't been the philosophy to date. We've permitted users to directly configure their `WebAppContext` in the pom without forcing them to create a context xml file so some users will be caught out. Also I think this is a fragile change: we add methods to `WebAppContext` in eg `ee10` that we want exposed to the user to configure but because `MavenWebAppContext` does not inherit from `WebAppContext` then they never get found by the ide refactoring tools. The whole point of the plugin is to configure and run a webapp and we want users to be able to do that as easily as possible.

Reply:
should be better now.
yup for sure it looks convenient to configure with the pom. But as I tried to explain previously if the users need to do some extra/sophisticated they will have to do it both in the pom and the jetty-web.xml as users will have to provide a jetty-web.xml when deploying the webapp in a production environment. So having to maintain 2 xml configuration with totally different syntax might not be very convenient.

Reply:
I'm not sure we're talking about the same things here. Currently users can configure their webapp in maven in any combination of the following:

+ the `WebAppContext` in the `pom.xml`
+ the `WebAppContext` via a normal context xml file
+ the `WebAppContext` via a normal `WEB-INF/jetty-web.xml` file

Reply:
we do :) 
But as you just described with configuration in `pom.xml` which have a totally different syntax from normal context xml or `WEB-INF/jetty-web.xml`, we just give users the option to configure the exact same thing in 2 different files with a totally different format whereas at the end of day they need only one format when deploying in production (e.g the jetty xml format).
And by the way this give us more work to have to maintain this extra layer `MavenWebAppContext`.
I would prefer having a very minimal set of options (port, resources, etc..) in the mojo and then if users need something more fancy they just use a jetty xml file as anyway they will need it to configure their webapp when deploying in a production environment (because they can not use Jetty Maven plugin in a production environment at least I hope not but who knows :)).
I don't mind if we keep all of those options but I find this a bit confusing for users and cause of possible mistakes because they have so many options. 
But hey it was just a thought I had when working on the code. if you don't want to change that's fine we do not have to change it.